internal/service/advancedcluster/resource_advanced_cluster_migration_test.go

```diff
@@ -104,6 +105,11 @@ func TestMigAdvancedCluster_shardedMigrationFromOldToNewSchema(t *testing.T) {
 				Config:            configShardedTransitionOldToNewSchema(orgID, projectName, clusterName, false),
 				Check:             checkShardedTransitionOldToNewSchema(false),
 			},
+			{
+				ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+				Config:                   configShardedTransitionOldToNewSchema(orgID, projectName, clusterName, true),
+				ExpectError:              regexp.MustCompile("SERVICE_UNAVAILABLE"),
```

Comment:
just to keep in mind, after service unavailable response migration is run asynchronously. We might have the risk following patch is done before migration is completed, so we should keep an eye on these tests and potentially add some delay before the 3rd step is run.

Reply:
do you have an idea of how we can add a delay in the test steps? I couldn't find anything relevant on this

Reply:
Thanks @EspenAlbert , @AgustinBettati does this work for you? lmk I'll update, afaik there's no built-in way to do this so I'm okay with the approach suggested by Espen.